ipwb/__main__.py

```diff
@@ -61,9 +61,10 @@ def checkArgs_replay(args):
     if suppliedIndexParameter:
         replay.start(cdxjFilePath=args.index, proxy=proxy)
     else:
-        # TODO: call replayParser.print_help()
-        print('Please supply a CDXJ file as an argument.')
-        sys.exit()
+        print('ERROR: An index file must be specified if not piping, e.g.,')
+        print('> ipwb replay /path/to/your/index.cdxj\n')
+
+        args.onError()
```

Comment:
Don't we also need `sys.exit()` after calling this `onError()` function?

Reply:
@ibnesayeed No. When we had it before it was unnecessary. The program exits after the help function is called.

Reply:
In that case we have many more places where `print_help` is being called and we have an explicit exit after that.

Reply:
I can confirm that `print_help` does not exit automatically. If it is behaving like that here, the reason might be something else further down the execution chain. Here is a sample test script I wrote to verify it.

```python
import argparse
parser = argparse.ArgumentParser()
parser.parse_args()
print("BEFORE HELP")
parser.print_help()
print("AFTER HELP")
```

If `print_help()` were to cause a forced exit then we will not see the `AFTER HELP` message, but that's not the case.

```
$ python help_test.py
BEFORE HELP
usage: main.py [-h]

optional arguments:
  -h, --help  show this help message and exit
AFTER HELP
```

Reply:
@ibnesayeed Per my comment above, I think the reason for the exit is that there is no further execution in the callee. I believe the exiting to be independent of the `print_help` invocation and simply that the callee's code is complete.

Reply:
Now I can see that the program terminates without an explicit exit here as nothing else is left to execute when the else branch is followed. This means we need an explicit `sys.exit()` here to ensure that the program behaves properly in situations where multiple commands are executed one after the other or an external program is watching for failures. Currently, the system would assume that everything is alright, though it is not.